Core/COEditingContext.m

```diff
@@ -369,6 +392,11 @@ - (void)undeletePersistentRoot: (COPersistentRoot *)aPersistentRoot
  */
 - (void)updateDeadRelationshipCacheForUndeletedPersistentRoot: (COPersistentRoot *)aPersistentRoot
 {
+	// FIXME:
```

Comment:
I think this code should be moved to -updateCrossPersistentRootReferencesToPersistentRoot:branch:isDeleted: and rewritten a bit since including the deleted branches as I did it seems wrong. For this last point, I could be overlooking something. 

-updateCrossPersistentRootReferencesToPersistentRoot:branch:isDeleted:
{	
	// more code before

	if (isDeleted)
		return;

	for (COBranch *branch in aPersistentRoot.branches)
	{
		[_deadRelationshipCache removePath: [COPath pathWithPersistentRoot: aPersistentRoot.UUID
		                                                            branch: branch.UUID]];
	}
	[_deadRelationshipCache removePath: [COPath pathWithPersistentRoot: aPersistentRoot.UUID]];
}

Is this correct or am I missing something?